auth/src/user.rs

```diff
@@ -207,6 +236,50 @@ impl UserManagement for UserManager {
             Err(Error::UserNotFound)
         }
     }
+
+    async fn insert_subscription(
+        &self,
+        name: &AccountName,
+        subscription_id: &str,
+        subscription_type: &models::user::SubscriptionType,
+        subscription_quantity: i32,
+    ) -> Result<(), Error> {
+        // Insert a new subscription. If the same type of subscription already exists, update the
+        // subscription id and quantity.
+        query(
+            r#"INSERT INTO subscriptions (subscription_id, account_name, type, quantity)
+            VALUES ($1, $2, $3, $4)
+            ON CONFLICT (account_name, type)
+            DO UPDATE SET subscription_id = EXCLUDED.subscription_id, quantity = EXCLUDED.quantity
+        "#,
+        )
```

Comment:
Oh, this is clever! I was thinking we would need another handler to increment the subscription, but this should work, and it's a lot simpler!